NullPointerException in DatabaseMetaData::getIndexInfo when querying the info for views

Try this in H2 2.0.202:

```java
try (Statement s = connection.createStatement()) {
    try {
        s.execute("create view v (i) as select 1");
        connection.getMetaData().getIndexInfo(null, "PUBLIC", "V", false, true);
    }
    finally {
        s.execute("drop view v");
    }
}
```

There is now a NullPointerException where there used to be an empty result set:

```
General error: "java.lang.NullPointerException: Cannot invoke ""java.util.ArrayList.iterator()"" because the return value of ""org.h2.table.Table.getIndexes()"" is null" [50000-202]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.message.DbException.toSQLException(DbException.java:367)
	at org.h2.message.TraceObject.logAndConvert(TraceObject.java:365)
	at org.h2.jdbc.JdbcDatabaseMetaData.getIndexInfo(JdbcDatabaseMetaData.java:264)
	at org.jooq.testscripts.JDBC.main(JDBC.java:43)
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.ArrayList.iterator()" because the return value of "org.h2.table.Table.getIndexes()" is null
	at org.h2.jdbc.meta.DatabaseMetaLocal.getIndexInfo(DatabaseMetaLocal.java:1263)
	at org.h2.jdbc.meta.DatabaseMetaLocal.getIndexInfo(DatabaseMetaLocal.java:1254)
	at org.h2.jdbc.JdbcDatabaseMetaData.getIndexInfo(JdbcDatabaseMetaData.java:262)
	... 1 more
```

--- a/h2/src/docsrc/html/features.html
+++ b/h2/src/docsrc/html/features.html
@@ -1157,7 +1157,7 @@ Java source code is generated as well. This simplifies reproducing problems. The
 12-20 20:58:09 jdbc[0]:
 /**/dbMeta3.getURL();
 12-20 20:58:09 jdbc[0]:
-/**/dbMeta3.getTables(null, "", null, new String[]{"TABLE", "VIEW"});
+/**/dbMeta3.getTables(null, "", null, new String[]{"BASE TABLE", "VIEW"});
 ...
 </pre>
 <p>

--- a/h2/src/main/org/h2/jdbc/meta/DatabaseMetaLocal.java
+++ b/h2/src/main/org/h2/jdbc/meta/DatabaseMetaLocal.java
@@ -1258,63 +1258,67 @@ public final class DatabaseMetaLocal extends DatabaseMetaLocalBase {
         return result;
     }
 
-    private void getIndexInfo(Value catalogValue, Value schemaValue, Table table, boolean unique, boolean approximate,
-            SimpleResult result, Database db) {
-        for (Index index : table.getIndexes()) {
-            if (index.getCreateSQL() == null) {
-                continue;
-            }
-            int uniqueColumnCount = index.getUniqueColumnCount();
-            if (unique && uniqueColumnCount == 0) {
-                continue;
-            }
-            Value tableValue = getString(table.getName());
-            Value indexValue = getString(index.getName());
-            IndexColumn[] cols = index.getIndexColumns();
-            ValueSmallint type = TABLE_INDEX_STATISTIC;
-            type: if (uniqueColumnCount == cols.length) {
-                for (IndexColumn c : cols) {
-                    if (c.column.isNullable()) {
-                        break type;
+    private void getIndexInfo(Value catalogValue, Value schemaValue, Table table, boolean unique,
+                                boolean approximate, SimpleResult result, Database db) {
+        ArrayList<Index> indexes = table.getIndexes();
+        if (indexes != null) {
+            for (Index index : indexes) {
+                if (index.getCreateSQL() == null) {
+                    continue;
+                }
+                int uniqueColumnCount = index.getUniqueColumnCount();
+                if (unique && uniqueColumnCount == 0) {
+                    continue;
+                }
+                Value tableValue = getString(table.getName());
+                Value indexValue = getString(index.getName());
+                IndexColumn[] cols = index.getIndexColumns();
+                ValueSmallint type = TABLE_INDEX_STATISTIC;
+                type:
+                if (uniqueColumnCount == cols.length) {
+                    for (IndexColumn c : cols) {
+                        if (c.column.isNullable()) {
+                            break type;
+                        }
                     }
+                    type = index.getIndexType().isHash() ? TABLE_INDEX_HASHED : TABLE_INDEX_OTHER;
                 }
-                type = index.getIndexType().isHash() ? TABLE_INDEX_HASHED : TABLE_INDEX_OTHER;
-            }
-            for (int i = 0, l = cols.length; i < l; i++) {
-                IndexColumn c = cols[i];
-                boolean nonUnique = i >= uniqueColumnCount;
-                if (unique && nonUnique) {
-                    break;
+                for (int i = 0, l = cols.length; i < l; i++) {
+                    IndexColumn c = cols[i];
+                    boolean nonUnique = i >= uniqueColumnCount;
+                    if (unique && nonUnique) {
+                        break;
+                    }
+                    result.addRow(
+                            // TABLE_CAT
+                            catalogValue,
+                            // TABLE_SCHEM
+                            schemaValue,
+                            // TABLE_NAME
+                            tableValue,
+                            // NON_UNIQUE
+                            ValueBoolean.get(nonUnique),
+                            // INDEX_QUALIFIER
+                            catalogValue,
+                            // INDEX_NAME
+                            indexValue,
+                            // TYPE
+                            type,
+                            // ORDINAL_POSITION
+                            ValueSmallint.get((short) (i + 1)),
+                            // COLUMN_NAME
+                            getString(c.column.getName()),
+                            // ASC_OR_DESC
+                            getString((c.sortType & SortOrder.DESCENDING) != 0 ? "D" : "A"),
+                            // CARDINALITY
+                            ValueBigint.get(approximate //
+                                    ? index.getRowCountApproximation(session)
+                                    : index.getRowCount(session)),
+                            // PAGES
+                            ValueBigint.get(index.getDiskSpaceUsed() / db.getPageSize()),
+                            // FILTER_CONDITION
+                            ValueNull.INSTANCE);
                 }
-                result.addRow(
-                        // TABLE_CAT
-                        catalogValue,
-                        // TABLE_SCHEM
-                        schemaValue,
-                        // TABLE_NAME
-                        tableValue,
-                        // NON_UNIQUE
-                        ValueBoolean.get(nonUnique),
-                        // INDEX_QUALIFIER
-                        catalogValue,
-                        // INDEX_NAME
-                        indexValue,
-                        // TYPE
-                        type,
-                        // ORDINAL_POSITION
-                        ValueSmallint.get((short) (i + 1)),
-                        // COLUMN_NAME
-                        getString(c.column.getName()),
-                        // ASC_OR_DESC
-                        getString((c.sortType & SortOrder.DESCENDING) != 0 ? "D" : "A"),
-                        // CARDINALITY
-                        ValueBigint.get(approximate //
-                                ? index.getRowCountApproximation(session)
-                                : index.getRowCount(session)),
-                        // PAGES
-                        ValueBigint.get(index.getDiskSpaceUsed() / db.getPageSize()),
-                        // FILTER_CONDITION
-                        ValueNull.INSTANCE);
             }
         }
     }

